docs-gen/content/rule_set/data_entry/data_unit_types.md

```diff
@@ -74,67 +74,132 @@ Vehicle.Powertrain.Transmission.ClutchWear:
+  domain: length
+  allowed-datatypes: ['numeric']
+```
+
+As `m` and `mm` are defined with the same domain it is expected that you can convert a value from `m`to `mm`.
+
+The `allowed-datatypes` attributes can be used to specify which VSS datatypes that can contain values of a particular unit.
+For most units the symbolic datatype `numeric` meaning any float or integer datatype can be used.
+As an example, both `uint8` and `float` can be used to represent a length value.
+If using `uint8` you will have range restrictions, but that might be acceptable for some signals.
+For some units more specific datatype restrictions are relevant. Some examples:
+
+* A date/time expressed in ISO 8061 format can only be represented in string
+* A UNIX Timestamp signal must be at least 32 bit unsigned to be able to handle date/time after year 2038.
+
+The `deprecation` keyword can be used to indicate that a specific unit may be removed in the future,
+Tooling shall preferably give a warning if a signal use a deprecated unit or the unit used belongs to a deprecated quantity. The reason should preferably list when and why the unit is deprecated, a hypothetical example is given below:
+
+```
+
+  inch:
+    definition: Distance measured in inches
+    unit: inch
+    domain: distance
+    deprecation: V5.0 replaced with in
+  in:
+    definition: Distance measured in inches
+    unit: inch
+    domain: distance
+```
+
+
+The unit syntax has recently been changed. To simplify transition to the new syntax it is recommended that
+tooling also supports unit files using the old syntax described below.
+
+```
+units:
+  [
+    <vss-unit-identifier>: # Typically unit abbreviation, like km/h or mm, but
+      label: <string> # Replaced with "unit"
+      description: <string> # Replaced with "definition"
+      domain: <string>
+  ]*
+```
+
+## Domain file syntax
+
+
+Defing domains is recommended, but currently optional for backward compatibility reasons.
+If tooling supports domain files it can verify that all units provided in unit files
+use defined domains.
+
+``
+
+  [
+    <vss-quantity-identifier>: # Identifier preferably taken from a standard, like ISO 80000
+      definition: <string>
+      [remark: <string>] # remark as defined in for example ISO 80000
+      [comment: <string>]
+      [deprecation: <reason>]
+  ]*
+
+```
+
+
+The VSS list of domains for the standard catalog exists in [domains.yaml](https://github.com/COVESA/vehicle_signal_specification/blob/master/spec/domains.yaml).
+
+## Defining custom units
+
+It is possible to define custom units in a unit file.
+Assume for instance you want to have a signal showing remaining range in [furlong](https://en.wikipedia.org/wiki/Furlong).
+Then you could add an additional unit `furlong`. No need to specify `unit` or `symbol` as they equals the default (i.e. "furlong").
+As this is unit not commonly used and not described in any standards it might be relevant to describe how it can be converted to other units.
```

Comment:
```suggestion
As this is unit not commonly used and not described in any standards, it might be relevant to describe how it can be converted to other units.
```